src/index.cpp

```diff
@@ -2804,10 +2631,18 @@ void Index<T, TagT, LabelT>::reposition_points(uint32_t old_location_start, uint
     // integer arithmetic rules.
     const uint32_t location_delta = new_location_start - old_location_start;
 
+    std::vector<location_t> updated_neighbours_location;
     for (uint32_t i = 0; i < _max_points + _num_frozen_pts; i++)
-        for (auto &loc : _final_graph[i])
+    {
+        auto &i_neighbours = _graph_store->get_neighbours((location_t)i);
```

Comment:
why is all this equivalent code not there in main?

Reply:
In main we were directly updating the neighbour_id's. But now we use get_neighbours() and its a const reference. So we have to copy it then augment it then set it.